### ISSUE TITLE
Integrated terminal slow to render output

### ISSUE THREAD

**Issue by iansimonson** [microsoft/vscode#159476]
Type: <b>Performance Issue</b>

In the past couple months the integrated terminal's rendering performance has become incredibly slow when there is a decent to large amount of output to the point it's essentially unusable for my day to day work. As shown below text rendering times can be 3x slower or more than using an external terminal. At first I thought it was just due to an increase in data given I generally tail a lot of log files however it affects my normal development actions like running `ctest -j30` with a large number of unittest binaries. This occurs even with no extensions enabled other than vscode-remote (necessary).

It seems to be purely a rendering issue. I measured execution time of running the mentioned tests using a desktop stopwatch and compared the result to `/usr/bin/time`. In powershell/wt/cmd the tests execute in 11.8 seconds. In the integrated terminal it takes **39.7** seconds. However, if I run the unit tests in the integrated terminal **and hide the terminal window** it will take a similar 12-13 seconds. /usr/bin/time reports 11.27-11.5 seconds for all cases.

Additionally, if I have a separate powershell/wt connected to the same tmux session, the integrated terminal will render at relatively normal speed (visually it seems like it detects the lines are rendered elsewhere and skips ahead to keep up).

Other notes:
My setup is vscode remote and tmux on the remote machine. I have been using this daily for the past two years. I compared with other terminals (powershell, cmd, wt, integrated) to rule out network latency issues. I'm not sure exactly which version I started noticing the performance regression but it was probably sometime around April-May.

EDIT: the unit tests above produce 1669 lines of text total. cat-ing a capture of that runs in 1-2 seconds on integrated vs < 1 second elsewhere. So it seems to be an issue even when the amount of data isn't massive.

VS Code version: Code 1.70.2 (e4503b30fc78200f846c62cf8091b76ff5547662, 2022-08-16T05:35:13.448Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Remote OS version: Linux x64 4.15.0-29-generic
Remote OS version: Linux x64 4.15.0-29-generic

<details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-ssh | ms- | 0.84.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30542924
vsaa593cf:30376535
pythonvs932:30410667
wslgetstarted:30449410
cppdebug:30492333
vscaat:30438848
pylanb8912:30545647
vsclangdf:30486550
c4g48928:30535728
d7fjc500:30543455
dsvsc012:30540252
azure-dev_surveyone:30548225
i497e931:30552240

```

</details>

**Comment by meganrogge** [comment]
Hi and thanks for all of the info in your report. Could you pls provide your `settings.json`?

**Comment by iansimonson** [comment]
Note: The GPU acceleration setting was from attempting to fix the problem locally by forcing gpu acceleration. The problem exists whether that's in the `settings.json` or not.

User settings.json:
<details>

```
{
    "remote.SSH.remotePlatform": {
        "sf-isimonson": "linux"
    },
    "workbench.colorTheme": "Dare (rainglow)",
    "workbench.editorAssociations": {
        "*.ipynb": "jupyter-notebook"
    },
    "workbench.sideBar.location": "right",
    "terminal.integrated.inheritEnv": false,
    "editor.minimap.enabled": false,
    "files.autoSave": "afterDelay",
    "terminal.integrated.commandsToSkipShell": [
        "-workbench.action.quickOpen",
        "-workbench.action.terminal.sendSequence",
        "-workbench.action.togglePanel",
        "-workbench.action.terminal.focusFind",
        "-editor.action.clipboardPasteAction",
        "-Terminal: Paste into Active Terminal",
        "-workbench.action.terminal.paste",
        "-workbench.action.quickOpenView"
    ],
    "terminal.integrated.allowChords": false,
    "clangd.onConfigChanged": "restart",
    "notebook.cellToolbarLocation": {
        "default": "right",
        "jupyter-notebook": "left"
    },
    "editor.renderWhitespace": "trailing",
    "editor.fontFamily": "'Inconsolata SemiExpanded Medium',Consolas, 'Courier New', monospace",
    "workbench.editor.navigationScope": "editorGroup",
    "terminal.integrated.enableMultiLinePasteWarning": false,
    "window.zoomLevel": -1,
    "terminal.integrated.gpuAcceleration": "on"
}
```
</details>

Remote settings:
<details>

```
{
    "C_Cpp.clang_format_path": "/home/isimonson/.local/llvm/latest/bin/clang-format",
    "C_Cpp.autocomplete": "Disabled",
    "C_Cpp.formatting": "Disabled",
    "C_Cpp.errorSquiggles": "Disabled",
    "C_Cpp.intelliSenseEngine": "Disabled",
    "clangd.path": "/home/isimonson/.local/llvm/latest/bin/clangd",
    "python.formatting.provider": "black",
    "cmake.generator": "Ninja",
    "python.experiments.enabled": false,
    "python.formatting.blackArgs": [
        "--line-length=100"
    ],
    "python.linting.flake8Args": [
        "--max-line-length=100",
        "--extend-ignore=E203"
    ],
    "remote.autoForwardPortsSource": "output",
    "cmake.clearOutputBeforeBuild": false,
    "cmake.configureOnEdit": false,
    "ols.server.path": "/home/isimonson/.local/bin/ols"
}
```

</details>

**Comment by Tyriar** [comment]
I fixed some performance regressions in the terminal renderers over the weekend actually, I expect this to be fixed in 1.72

**Comment by Tyriar** [comment]
Let's track in https://github.com/microsoft/vscode/issues/157432
